src/sctools/platform.py

```diff
@@ -629,3 +629,318 @@ def attach_barcodes(cls, args=None):
         cls._tag_bamfile(args.u2, args.output_bamfile, tag_generators)
 
         return 0
+
+
+class BarcodePlatform(GenericPlatform):
+    """Command Line Interface for extracting and attaching barcodes with specified positions
+     generalizing TenXV2 attach barcodes
+
+    Sample, cell and/or molecule barcodes can be extracted and attached to an unmapped bam when the
+    corresponding barcode's start position and and length are provided. The sample barcode is extracted
+    from the index i7 fastq file and the cell/molecule barcode is extracted from the r1 fastq file
```

Comment:
```suggestion
    from the index i7 fastq file and the cell and molecule barcodes are extracted from the r1 fastq file
```

Reply:
I will create an issue and link back to this PR. Thank you for the review!